addons/lunch/models/lunch.py

```diff
@@ -0,0 +1,296 @@
+
+    category_id = fields.Many2one('lunch.product.category', string='Product Category',
+                                  related='product_id.category_id', readonly=True, store=True)
+
+    date = fields.Date(string='Date', related='order_id.date', readonly=True, store=True)
+
+    supplier = fields.Many2one('res.partner', string='Supplier', related='product_id.supplier',
+                               readonly=True, store=True)
+
+    user_id = fields.Many2one('res.users', string='User', related='order_id.user_id',
+                              readonly=True, store=True)
+
+    note = fields.Text('Note')
+    price = fields.Float('Price')
+    state = fields.Selection([('new', 'New'),
+                              ('confirmed', 'Received'),
+                              ('ordered', 'Ordered'),
+                              ('cancelled', 'Cancelled')],
+                             'Status', readonly=True, select=True, default='new')
+
+    cashmove = fields.One2many('lunch.cashmove', 'order_id', 'Cash Move')
+    currency_id = fields.Many2one('res.currency', related='order_id.currency_id')
+
+
+class lunch_product(models.Model):
+    """
+    lunch product
+    """
+    _name = 'lunch.product'
+    _description = 'lunch product'
+
+    name = fields.Char('Product', required=True)
+    category_id = fields.Many2one('lunch.product.category', 'Category', required=True)
+    description = fields.Text('Description')
+    price = fields.Float('Price', digits=(16, 2))
+    supplier = fields.Many2one('res.partner', 'Supplier')
+
+
+class lunch_product_category(models.Model):
+    """
+    lunch product category
+    """
+    _name = 'lunch.product.category'
+    _description = 'lunch product category'
+
+    # such as PIZZA, SANDWICH, PASTA, CHINESE, BURGER, ...
+    name = fields.Char('Category', required=True)
+
+
+class lunch_cashmove(models.Model):
+    """
+    lunch cashmove => order or payment
+    """
+    _name = 'lunch.cashmove'
+    _description = 'lunch cashmove'
+
+    @api.multi
+    def name_get(self):
+        return [(cashmove.id, _("Lunch Cashmove #%d") % cashmove.id) for cashmove in self]
+
+    user_id = fields.Many2one('res.users', 'User Name', required=True,
+                              default=lambda self: self.env.uid)
+
+    date = fields.Date('Date', required=True, default=fields.Date.context_today)
+
+    amount = fields.Float('Amount', required=True) # Can be positive or negative
+
+    description = fields.Text('Description') # Can be an order or a payment
+
+    order_id = fields.Many2one('lunch.order.line', 'Order', ondelete='cascade')
+
+    state = fields.Selection([('order', 'Order'), ('payment', 'Payment')],
+                             'Is an order or a Payment', default='payment')
+
+
+class lunch_alert(models.Model):
+    """
+    lunch alert
+    """
+    _name = 'lunch.alert'
+    _description = 'Lunch Alert'
+
+    @api.multi
+    def name_get(self):
```

Comment:
I think if the message itself is explicit enough to avoid having a real name. Otherwise, you could have situation like:
- name : "Daily closing of MySuperLunchFactory"
- message : "MySuperLunchFactory is closed on Monday"

To be discussed...